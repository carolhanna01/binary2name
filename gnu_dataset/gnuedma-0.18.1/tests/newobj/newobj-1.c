/*
 * EDMA: Entorno de Desarrollo Modular y Abierto
 * Object Oriented and Componetware Framework
 * Copyright (C) 1998, 2002, 2003, 2004, 2005, 2006, 2007, 2008, 2010
 *    David Martínez Oliveira
 *
 * This file is part of EDMA.
 *
 * EDMA is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Lesser General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * EDMA is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with EDMA.  If not, see <http://www.gnu.org/licenses/>.
 */

#include <stdio.h>
#include <edma.h>

int main (int argc, char *argv[])
{
  OBJID		id, id1;
  
  EDMAInit();

  /* Simple Object Creation */
  id = edma_new_obj ("HELLO_WORLD");
  edma_obj_report (id);
  edma_free_obj (id);


  /* Single Static Inheritance*/
  id = edma_new_obj ("GRADUATE");
  edma_obj_report (id);
  edma_free_obj (id);

  /* SIU PROXY does exists*/
  id = edma_new_obj ("RUNABLE:HELLO_WORLD");
  edma_met3 (id, "say");
  edma_obj_report (id);
  edma_free_obj (id);

  /* SIU PROXY doesn't exists*/
  id = edma_new_obj ("UNKNOWN_PROXY:HELLO_WORLD");
  edma_obj_report (id);
  edma_free_obj (id);

  EDMAEnd();
}
