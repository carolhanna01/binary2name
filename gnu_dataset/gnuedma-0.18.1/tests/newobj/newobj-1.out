
GNU/EDMA 0.16.4 build [Jan 30 2010][Beta][0.16.4] starting up...
EDMA Startup process... OK
(HELLO_WORLD)  [Constructor]
==OBJECT REPORT==================================================
Object (0) of class (HELLO_WORLD)
Top Level Object. Object haven't father
No PseudoFather information
Object has 0 superobjects
Object has 0 subobjects
Virtual Methods in object
==END OBJECT REPORT==========================================
(HELLO_WORLD)  [Destructor]
==OBJECT REPORT==================================================
Object (0) of class (GRADUATE)
Top Level Object. Object haven't father
No PseudoFather information
Object has 1 superobjects
    Id : (PERSON) Class : (PERSON) Class attached Object: (PERSON)
Object has 0 subobjects
Virtual Methods in object
==END OBJECT REPORT==========================================
(RUNABLE)Creating object of class HELLO_WORLD
(HELLO_WORLD)  [Constructor]
(RUNABLE)Creating thread for method say
[RUNABLE] Invoking threaded method say on object 1
(HELLO_WORLD)  [say] -------------------------------
(HELLO_WORLD)  [say] Hello world!!!!
(HELLO_WORLD)  [say] Property num : 0
(HELLO_WORLD)  [say] Property str : (null)
(HELLO_WORLD)  [say] -------------------------------
[edma_thread_unregister] Unregistering thread 0xb7d08b90
======> Index 1 of 2[edma_thread_unregister] Thread 0xb7d08b90 found at index 1
==OBJECT REPORT==================================================
Object (0) of class (RUNABLE)
Top Level Object. Object haven't father
No PseudoFather information
Object has 0 superobjects
Object has 0 subobjects
Virtual Methods in object
==END OBJECT REPORT==========================================
(HELLO_WORLD)  [Destructor]
[EDMA_EXCEPTION] <1> :: [edma_obj_report] Invalid Object Identifier: -1
[EDMA_EXCEPTION] <1> :: [edma_free_obj] Invalid Object Identifier: -1
---------------------[EDMAEnd]----------------------------------
No more process in GNU EDMA. Freeing All Resources...
Cleanup done!!
